pkg/skaffold/kubernetes/status/status_check.go

```diff
@@ -71,40 +75,65 @@ type Config interface {
 	GetNamespaces() []string
 	StatusCheckDeadlineSeconds() int
 	Muted() config.Muted
-}
-
-// Checker waits for the application to be totally deployed.
-type Checker interface {
-	Check(context.Context, io.Writer) error
+	StatusCheck() (*bool, error)
 }
 
 // statusChecker runs status checks for pods and deployments
-type statusChecker struct {
+type Checker struct {
```

Comment:
since we're making all these changes now, I wonder if we should rename this to `Monitor`. then we have `log`, `debug`, `monitor`, `access` as the 4 actions to perform on deployed resources. I think `monitor` flows a little better than `status check`

Reply:
done